lnwallet/channel.go

```diff
@@ -2151,22 +2170,74 @@ func (lc *LightningChannel) closeObserver(channelCloseNtfn *chainntnfs.SpendEven
 			"broadcast!!!", lc.channelState.FundingOutpoint,
 			remoteStateNum)
 
+		if err := lc.channelState.MarkBorked(true); err != nil {
+			lc.Unlock()
+			walletLog.Errorf("unable to mark channel as borked: %v", err)
+			return
+		}
+
+		settledBalance := lc.channelState.LocalCommitment.LocalBalance.ToSatoshis()
+		closeSummary := channeldb.ChannelCloseSummary{
+			ChanPoint:      lc.channelState.FundingOutpoint,
+			ChainHash:      lc.channelState.ChainHash,
+			ClosingTXID:    *commitSpend.SpenderTxHash,
+			RemotePub:      lc.channelState.IdentityPub,
+			Capacity:       lc.Capacity,
+			SettledBalance: settledBalance,
+			CloseType:      channeldb.BreachClose,
+			IsPending:      true,
```

Comment:
While were at it, we can also write the height that channel got borked at. This way, we can use this height when requesting confirmations in the breach arbiter, etc.

Reply:
great idea!